src/utils.cpp

```diff
@@ -2002,21 +2019,166 @@ void NodeCounter::operator -= (const NodeCounter& o)
+}
 
+handle SyncConfig::getLocalFingerprint() const
+{
+    return mLocalFingerprint;
+}
 
+const std::vector<std::string>& SyncConfig::getRegExps() const
+{
+    return mRegExps;
+}
 
+bool SyncConfig::isUpSync() const
+{
+    return mSyncType & TYPE_UP;
+}
 
+bool SyncConfig::isDownSync() const
+{
+    return mSyncType & TYPE_DOWN;
+}
 
+bool SyncConfig::syncDeletions() const
+{
+    switch (mSyncType)
+    {
+        case TYPE_UP: return mSyncDeletions;
+        case TYPE_DOWN: return mSyncDeletions;
+        case TYPE_TWOWAY: return true;
+    }
+    assert(false);
+    return true;
+}
 
+bool SyncConfig::forceOverwrite() const
+{
+    switch (mSyncType)
+    {
+        case TYPE_UP: return mForceOverwrite;
+        case TYPE_DOWN: return mForceOverwrite;
+        case TYPE_TWOWAY: return false;
+    }
+    assert(false);
+    return false;
+}
 
+std::string SyncConfig::serialize() const
+{
+    std::string data;
+    CacheableWriter writer{data};
+    writer.serializebool(mActive);
+    writer.serializestring(mLocalPath);
+    writer.serializehandle(mRemoteNode);
+    writer.serializefsfp(mLocalFingerprint);
+    writer.serializeu32(static_cast<uint32_t>(mRegExps.size()));
+    for (const auto& regExp : mRegExps)
+    {
+        writer.serializestring(regExp);
+    }
+    writer.serializei64(mSyncType);
+    writer.serializebool(mSyncDeletions);
+    writer.serializebool(mForceOverwrite);
+    writer.serializeexpansionflags();
+    return data;
+}
 
+std::unique_ptr<SyncConfig> SyncConfig::unserialize(const std::string& data)
+{
+    bool active;
+    std::string localPath;
+    handle remoteNode;
+    fsfp_t fingerprint;
+    uint32_t regExpCount;
+    std::vector<std::string> regExps;
+    int64_t syncType;
+    bool syncDeletions;
+    bool forceOverwrite;
 
+    CacheableReader reader{data};
+    if (!reader.unserializebool(active))
+    {
+        return {};
+    }
+    if (!reader.unserializestring(localPath))
+    {
+        return {};
+    }
+    if (!reader.unserializehandle(remoteNode))
+    {
+        return {};
+    }
+    if (!reader.unserializefsfp(fingerprint))
+    {
+        return {};
+    }
+    if (!reader.unserializeu32(regExpCount))
+    {
+        return {};
+    }
+    for (uint32_t i = 0; i < regExpCount; ++i)
+    {
+        std::string regExp;
+        if (!reader.unserializestring(regExp))
+        {
+            return {};
+        }
+        regExps.push_back(std::move(regExp));
+    }
+    if (!reader.unserializei64(syncType))
+    {
+        return {};
+    }
+    if (!reader.unserializebool(syncDeletions))
+    {
+        return {};
+    }
+    if (!reader.unserializebool(forceOverwrite))
+    {
+        return {};
+    }
 
+    auto syncConfig = std::unique_ptr<SyncConfig>{new SyncConfig{std::move(localPath),
+                    remoteNode, fingerprint, std::move(regExps),
+                    static_cast<Type>(syncType), syncDeletions, forceOverwrite}};
+    syncConfig->setActive(active);
```

Comment:
The `active` property of a `SyncConfig` should be persisted, shouldn't it? We don't want to remove the `SyncConfig` just because the user has disabled the sync temporary.

Reply:
Oh, my bad. Sorry!